apps/protoc-gen-openapi/plugin_test.go

```diff
@@ -17,68 +17,67 @@ package main
+}{
+	{name: "Google Lbrary example", path: "examples/google/example/library/v1/", protofile: "library.proto"},
+	{name: "Body mapping", path: "examples/tests/bodymapping/", protofile: "message.proto"},
+	{name: "Map fields", path: "examples/tests/mapfields/", protofile: "message.proto"},
+	{name: "Path params", path: "examples/tests/pathparams/", protofile: "message.proto"},
+	{name: "Protobuf types", path: "examples/tests/protobuftypes/", protofile: "message.proto"},
+	{name: "JSON options", path: "examples/tests/jsonoptions/", protofile: "message.proto"},
 }
 
-func TestBodyMappingOpenAPI(t *testing.T) {
-	var err error
-	// Run protoc and the protoc-gen-openapi plugin to generate an OpenAPI spec.
-	err = exec.Command("protoc",
-		"-I", "../../",
-		"-I", "../../third_party",
-		"-I", "examples",
-		"examples/tests/bodymapping/message.proto",
-		"--openapi_out=.").Run()
-	if err != nil {
-		t.Fatalf("protoc failed: %+v", err)
+func TestOpenAPIProtobufNaming(t *testing.T) {
+	for _, tt := range openapiTests {
+		t.Run(tt.name, func(t *testing.T) {
+			// Run protoc and the protoc-gen-openapi plugin to generate an OpenAPI spec.
+			err := exec.Command("protoc",
+				"-I", "../../",
+				"-I", "../../third_party",
+				"-I", "examples",
+				path.Join(tt.path, tt.protofile),
+				"--openapi_out=.").Run()
+			if err != nil {
+				t.Fatalf("protoc failed: %+v", err)
+			}
+			// Verify that the generated spec matches our expected version.
+			err = exec.Command("diff", "openapi.yaml", path.Join(tt.path, "openapi.yaml")).Run()
+			if err != nil {
+				t.Fatalf("Diff failed: %+v", err)
+			}
+			// if the test succeeded, clean up
+			os.Remove("openapi.yaml")
+		})
 	}
-	// Verify that the generated spec matches our expected version.
-	err = exec.Command("diff", "openapi.yaml", "examples/tests/bodymapping/openapi.yaml").Run()
-	if err != nil {
-		t.Fatalf("Diff failed: %+v", err)
-	}
-	// if the test succeeded, clean up
-	os.Remove("openapi.yaml")
 }
 
-func TestMapFieldsOpenAPI(t *testing.T) {
-	var err error
-	// Run protoc and the protoc-gen-openapi plugin to generate an OpenAPI spec.
-	err = exec.Command("protoc",
-		"-I", "../../",
-		"-I", "../../third_party",
-		"-I", "examples",
-		"examples/tests/mapfields/message.proto",
-		"--openapi_out=.").Run()
-	if err != nil {
-		t.Fatalf("protoc failed: %+v", err)
-	}
-	// Verify that the generated spec matches our expected version.
-	err = exec.Command("diff", "openapi.yaml", "examples/tests/mapfields/openapi.yaml").Run()
-	if err != nil {
-		t.Fatalf("Diff failed: %+v", err)
+func TestOpenAPIJSONNaming(t *testing.T) {
+	for _, tt := range openapiTests {
+		t.Run(tt.name, func(t *testing.T) {
+			// Run protoc and the protoc-gen-openapi plugin to generate an OpenAPI spec with JSON naming.
+			err := exec.Command("protoc",
+				"-I", "../../",
+				"-I", "../../third_party",
+				"-I", "examples",
+				path.Join(tt.path, tt.protofile),
+				"--openapi_out=json=true,version=1.2.3:.").Run()
```

Comment:
I have been trying to find the arguments parser in the protoc source, but it's all over the place and after spending an hour, I wasn't able to find the exact place where the actual argument value parsing is done...
But it seems like it will interpret an argument as a string, that ends if it finds a comma.
So much like a low level parser - find bytes and pass them along, without interpreting the meaning (except for commas).

So from what I can deduct: Seen from the perspective of protoc, it is really not meant as a way of passing in configuration values - more like options to change behaviour.
This is the same for both *_out=param=value,... and *_opt=param=value.

@timburks what are your thoughts on this? Keep? Drop? Find another way?